pkg/operation/botanist/health_check.go

```diff
@@ -488,12 +488,19 @@ func (b *HealthChecker) CheckMonitoringControlPlane(
 		return exitCondition, nil
 	}
 
-	statefulSetList, err := statefulSetLister.StatefulSets(namespace).List(monitoringSelector)
+	requiredMonioringStatefulSets := sets.NewString(common.PrometheusStatefulSetName)
+	if alertingSecret := botanist.GetSecretKeysOfRole(common.GardenRoleAlertingSMTP); len(alertingSecret) > 0 {
```

Comment:
This code is duplicated multiple times, what about computing it once and store it on the `Shoot` object, similar to https://github.com/gardener/gardener/blob/master/pkg/operation/shoot/types.go#L34?